etl/migrations/20251127090000_schema_change_messages.sql

```diff
@@ -0,0 +1,173 @@
+-- Schema change logical messages (DDL)
+-- Adds helpers and trigger to emit logical decoding messages when tables change.
+
+create or replace function etl.describe_table_schema(
+    p_table oid
+) returns table (
+    name text,
+    type_oid oid,
+    type_name text,
+    type_modifier int,
+    ordinal_position int,
+    primary_key_ordinal_position int,
+    nullable boolean
+)
+language plpgsql
+stable
+as
+$$
+declare
+    v_query text;
+begin
+    v_query := format(
+        'with direct_parent as (
+            select i.inhparent as parent_oid
+            from pg_inherits i
+            where i.inhrelid = %1$s
+            limit 1
+        ),
+        primary_key as (
+            select x.attnum, x.n as position
+            from pg_constraint con
+            join unnest(con.conkey) with ordinality as x(attnum, n) on true
```

Comment:
Are we doing this to then rebuild the constraint on the downstream?
Perhaps worth to consider `pg_get_constraintdef`?